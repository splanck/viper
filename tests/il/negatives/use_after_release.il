il 0.1.2
extern @rt_arr_i32_len(ptr) -> i64
extern @rt_arr_i32_release(ptr) -> void

func @main() -> i64 {
entry:
  %slot = alloca 8
  store ptr, %slot, null
  %handle = load ptr, %slot
  call @rt_arr_i32_release(%handle)
  %len = call @rt_arr_i32_len(%handle)
  ret %len
}
