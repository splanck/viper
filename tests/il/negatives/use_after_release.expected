error: main:entry: %2 = call %t1: use after release of %1
